src/vs/workbench/contrib/notebook/browser/notebookEditorWidget.ts

```diff
@@ -1079,7 +1079,7 @@ export class NotebookEditorWidget extends Disposable implements INotebookEditorD
 					return;
 				}
 
-				if (this._dimension) {
+				if (this._dimension && this._isVisible) {
 					if (sizeDelta > 0) { // delta > 0 ==> sticky is growing, cell list shrinking
 						this.layout(this._dimension);
```

Comment:
Should the visibility check be inside `this.layout` also?

Reply:
@roblourens yes, there are multiple places doing this redundant check and I plan to move them into `layout` during debt, and do some thorough testing to ensure I don't miss anything.